types/index.d.ts

```diff
@@ -0,0 +1,424 @@
+import { ZwavejsServer } from '@zwave-js/server'
+import { EventEmitter } from 'events'
+import { MqttClient as Client, IClientPublishOptions } from 'mqtt'
+import { Socket } from 'net'
+import {
+  Association,
+  CommandClass,
+  InterviewStage,
+  NodeStatus,
+  ValueID,
+  ValueType,
+  ZWaveNode,
+  ZWaveOptions
+} from 'zwave-js'
+
+export type Z2MValueIdState = {
```

Comment:
Normally, you'd use an `interface` for these kinds of types (also supports type inheritance with `extends`).

Reply:
Couldn't use the `&` operator instead of the `extends` with types?

Reply:
BTW this is just a temporary fix, I'm looking forward to switch to typescript completely. Also I think that using a namespace here (instead of prefix names with `Z2M` would be better`

Reply:
& is fine to combine types. It is just recommended to use interfaces for these simple object types without conditionals.

Reply:
So that also applies to TS later